src/Lumper.UI/Updater/Updater.cs

```diff
@@ -0,0 +1,218 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.Http;
+using System.Net;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Text.Json.Nodes;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Lumper.UI.ViewModels;
+using MessageBox.Avalonia.Enums;
+using MessageBox.Avalonia;
+using Avalonia.Controls;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.IO;
+
+namespace Lumper.UI.Updater
+{
+    internal class Updater
+    {
+        //struct for deserializing JSON objects
+        private struct Asset
+        {
+            [JsonProperty("browser_download_url")]
+            public string browser_download_url;
+            [JsonProperty("name")]
+            public string name;
+        }
+        //struct for deserializing JSON objects
+        private struct GHUpdate
+        {
+            [JsonProperty("tag_name")]
+            public string tag_name;
+            [JsonProperty("assets")]
+            public Asset[] assets;
+
+
+        }
+        //Major/Minor/Patch format
+        public class MMP
+        {
+            public int major;
+            public int minor;
+            public int patch;
+            public MMP(int Major, int Minor, int Patch)
+            {
+                major = Major;
+                minor = Minor;
+                patch = Patch;
+            }
+        }
+
+        private static MMP GetMMPVersion(string s)
+        {
+            //match pattern of xx.yy.zz
+            Match match = Regex.Match(s, "[0-9]+\\.[0-9]+\\.[0-9]+");
+            MMP version = new MMP(0, 0, 0);
+            if (match.Success)
+            {
+                MatchCollection currentVersion = Regex.Matches(match.ToString(), "[0-9]+");
+
+
+                if (int.TryParse(currentVersion[0].ToString(), out version.major)
+                    && int.TryParse(currentVersion[1].ToString(), out version.minor)
+                    && int.TryParse(currentVersion[2].ToString(), out version.patch))
+                {
+                }
+                else
+                {
+                    throw new Exception("Could not parse Major/Minor/Patch version.");
+                }

```

Comment:
You've already verified that this regex fits this pattern so the TryParse should always be good - if the pattern was bad we'd have just returned a version `0.0.0`. So I would just use `int.Parse`.

I realise I've completely torn this method apart, I'm just a regex nerd and spent some time learning C# regexes lately (C# regex support is [*really* good](https://www.youtube.com/watch?v=ptKjWPC7pqw)!)

Also, I'd make this a static method on `Version` (previously MMP) just called `Parse`
Here's how I'd write the whole thing, though haven't had time to test and I might be getting the capture groups stuff slightly wrong:
```cs
private static Version Parse(string s)
{
  Match match = VersionRegex.Match(s);

  if (!match.Success || match,Captures.Count != 3)
    throw new InvalidDataException($"Failed to parse version for string {s}");

  return new MMP(
    int.Parse(match.Captures[0].Value),
    int.Parse(match.Captures[1].Value)
    int.Parse(match.Captures[2].Value)
  )
}

[GeneratedRegex(@"^(\d+)\.(\d+)\.(\d+)$")]
private static partial Regex VersionRegex();
```